src/bootstrap/test.rs

```diff
@@ -794,7 +807,9 @@ impl Step for RustdocGUI {
     const ONLY_HOSTS: bool = true;
 
     fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
-        run.path("src/test/rustdoc-gui")
+        let builder = run.builder;
+        let run = run.path("src/test/rustdoc-gui");
+        run.default_condition(builder.config.nodejs.is_some())
```

Comment:
I believe we had discussed an alternative formulation here:

* use check_if_browser_ui_test_is_installed as the default condition
* if the step is running, panic if anything is not installed (browser-ui-test, node, etc.)